### ISSUE TITLE
Specific Exceptions: Adapting the interfaces

### ISSUE THREAD

**Issue by felixdivo** [hardbyte/python-can#1046]
Follow up to #1025. All interfaces need to be changed:

- [x] ics_neovi -> @felixdivo #1073; fix in #1080
- [x] ixxat -> @felixdivo already done in the original #1025
- [x] kvaser -> @felixdivo #1074
- [x] neousys -> @felixdivo #1076
- [x] pcan -> @felixdivo #1152
- [x] seeedstudio -> @felixdivo #1090
- [x] serial -> @felixdivo #1088
- [x] socketcan -> @felixdivo #1087
- [x] systec -> @felixdivo #1158
- [x] udp_multicast -> @felixdivo #1089
- [x] usb2can -> @felixdivo #1091
- [x] vector -> @felixdivo already started in the original #1025; fixed in #1052; completed in #1083
- [x] canalystii -> @projectgus #1127
- [x] cantact -> @felixdivo #1153
- [x] gs_usb -> @felixdivo #1168
- [x] iscan -> @felixdivo #1092
- [x] nixnet -> @felixdivo #1154
- [x] nican -> @felixdivo #1100
- [x] robotell -> @felixdivo #1155
- [x] slcan -> @felixdivo #1156 and #1166
- [x] virtual -> @felixdivo #1079

**Comment by zariiii9003** [comment]
@felixdivo can you summarize the necessary changes?

**Comment by felixdivo** [comment]
Sure: #1025 added subclass exceptions to `can.CanError`, which should now be used by the interfaces. One should go trough the interfaces one by one and replace any use of `can.CanError` by some more specific exception type (where applicable).

Does that help? :) If you want to help, just add your name or some PR link to the list above. I'll tackle this the coming week.

**Comment by felixdivo** [comment]
Reviewing all interfaces is quite a bit of work. Help if of course much appreciated.

**Comment by felixdivo** [comment]
THis is complete once all PRs which are linked in the issue text are merged.
